docs/yul.rst

```diff
@@ -336,248 +682,207 @@ We will use a destructuring notation for the AST nodes.
-TBD
+.. _opcodes:
+
+EVM Dialect
+-----------
+
+The default dialect of Yul currently is the EVM dialect parameterized
+with a version of the EVM. The only type available in this dialect
+is ``u256``, the 256-bit native type of the Ethereum Virtual Machine.
+Since it is the default type of this dialect, it can be omitted.
+
+The following table lists all builtin functions
+(depending on the EVM version) and provides a short description of the
+semantics of the function / opcode.
+This document does not want to be a full description of the Ethereum virtual machine,
+so please refer to a different document if you are interested in the precise semantics.
+
+Opcodes marked with ``-`` do not return a result and all others return exactly one value.
+Opcodes marked with ``F``, ``H``, ``B``, ``C`` or ``I`` are present since Frontier, Homestead,
+Byzantium, Constantinople or Istanbul, respectively.
+
+In the following, ``mem[a...b)`` signifies the bytes of memory starting at position ``a`` up to
+but not including position ``b`` and ``storage[p]`` signifies the storage contents at slot ``p``.
+
+Since Yul manages local variables and control-flow,
+opcodes that interfere with these features are not available. This includes
+the ``dup`` and ``swap`` instructions as well as ``jump`` instructions and labels.
+
++-------------------------+-----+---+-----------------------------------------------------------------+
+| Instruction             |     |   | Explanation                                                     |
++=========================+=====+===+=================================================================+
+| stop()                  + `-` | F | stop execution, identical to return(0, 0)                       |
++-------------------------+-----+---+-----------------------------------------------------------------+
+| add(x, y)               |     | F | x + y                                                           |
++-------------------------+-----+---+-----------------------------------------------------------------+
+| sub(x, y)               |     | F | x - y                                                           |
++-------------------------+-----+---+-----------------------------------------------------------------+
+| mul(x, y)               |     | F | x * y                                                           |
++-------------------------+-----+---+-----------------------------------------------------------------+
+| div(x, y)               |     | F | x / y or 0 if y == 0                                            |
++-------------------------+-----+---+-----------------------------------------------------------------+
+| sdiv(x, y)              |     | F | x / y, for signed numbers in two's complement                   |
++-------------------------+-----+---+-----------------------------------------------------------------+
+| mod(x, y)               |     | F | x % y                                                           |
```

Comment:
Maybe we should also mention the behaviour for ``y = 0`` here, since it's not particularly well defined in general outside the EVM either.